core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -423,18 +432,35 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
     tasks.toMap
   }
 
+  private def getResources(resources: JList[Resource], taskCPUs: Int, taskMemory: Int)
+    : (List[Resource], List[List[Resource]], List[Long]) = {
+    val (afterCPUResources, cpuResourcesToUse) =
+      partitionResources(resources, "cpus", taskCPUs)
+    val (afterMemResources, memResourcesToUse) =
+      partitionResources(afterCPUResources.asJava, "mem", taskMemory)
+    // process port offers
+    val (resourcesWithoutPorts, portResources) = filterPortResources(afterMemResources)
+    val (afterPortResources, portResourcesToUse, portsToUse) =
+      partitionPorts(conf, portResources)
+    (resourcesWithoutPorts ++ afterPortResources,
```

Comment:
place each of the three tuple elements in the return on separate lines for readability